python/tvm/auto_scheduler/compute_dag.py

```diff
@@ -182,3 +187,11 @@ def hash_key(self):
 
         str_key = str_key.encode(encoding="utf-8")
         return hashlib.md5(str_key).hexdigest()
+
+    def __getstate__(self):
+        return {"compute": SaveJSON(self.compute), "sche": SaveJSON(self.sche)}
+
+    def __setstate__(self, state):
+        self.compute = LoadJSON(state["compute"])  # pylint: disable=assignment-from-no-return
```

Comment:
@comaniac Do we need to explicitly call `SaveJSON`, `LoadJSON` here?
I find that you don't call it in the `SearchTask::__getstate__`

Reply:
We don't call them in SearchTask because all its members have these two functions implemented correctly. This function provides the correct implementation for ConouteDAG and it will be called When processing ComputeDAG in SearchTask.

For Load/SaveJSON here, alternatively we can use pickle.loads/dumps and let them call Load/SaveJSON via Objects. However, this will introduce the dependency of pickle in this function, and will somehow trigger a bug in the unit test. You can test it by replacing Load/SaveJSON with pickle calls. In the unit test I add, you will find that after loading the DAG back, the placeholder A appears twice.